packages/top-app-bar/index.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import {
+  MDCTopAppBarFoundation,
+  MDCShortTopAppBarFoundation,
+} from '@material/top-app-bar';
+
+export default class TopAppBar extends React.Component {
+
+  foundation_ = null;
+
+  state = {
+    classList: new Set(),
+  };
+
+  get classes() {
+    const {classList} = this.state;
+    const {
+      shortCollapsed,
+      className,
+      short,
+      prominent,
+    } = this.props;
+
+    return classnames('mdc-top-app-bar', Array.from(classList), className, {
+      'mdc-top-app-bar--short': shortCollapsed || short,
+      'mdc-top-app-bar--short-collapsed': shortCollapsed,
+      'mdc-top-app-bar--prominent': prominent,
+    });
+  }
+
+  componentDidMount() {
+    this.initializeFoundation();
+  }
+
+  componentWillUnmount() {
+    this.foundation_.destroy();
+  }
+
+  initializeFoundation = () => {
+    if (this.props.short) {
+      this.foundation_ = new MDCShortTopAppBarFoundation(this.adapter);
+    } else {
+      this.foundation_ = new MDCTopAppBarFoundation(this.adapter);
+    }
+
+    this.foundation_.init();
+  }
+
+  // TODO: Move to a parent component
+  addClassesToElement(classes, element) {
+    // check if element passed is a user defined React Component
+    const isComponent = typeof element.type === 'function';
+    let elementInstance = element;
+
+    if (isComponent) {
+      // this is needed to get to render method of component
+      const Component = element.type;
+      const component = new Component();
```

Comment:
@lynnjepsen this is the only way I could come up with adding a class to a Component that is passed in (ie `<NavIcon>`). If they pass in a regular tag such as `<a >` or `<svg>`, then this if block is unnecessary.  I've never seen this pattern, nor could I find anyone referencing it is a bad idea or a good idea.

I think since we are a library of components we will be doing some wild and potentially controversial things. Let me know what you think. 

*Also I feel like if I use the keyword `new`, I should probably be deleting this after I use it. But before I dive into that I wanted to see what you thought.

Reply:
Why wouldn't you defer to composition for this? if I have an `AppBar` with `NavIcon`s, as a user I would do this:

```
<TOP APP BAR COMPONENT STUFF>
render() {
  return (
    <TopAppBar>
      <NavIcon className={`mdc-nav-icon ${conditional classes}`}></NavIcon
    </TopAppBar>
  );
}
```